git forking - closes #88 (#92)
bob-crutchley/notes: topics/git/modules/forking/README.md

> # Forking

Reviewer: ```suggestion
<!--PROPS
{
	"estTime": 10
}
-->
# Forking
```

Forking a repository allows to freely experiment with the existing code base without the fear of breaking the project.
Another use case is to contribute to the project by adding additional functionality. 

## Proposing a change

In a scenario where you are using someones project and you find a bug, instead of submitting an issue about it you can try 
fixing it. The steps would be:
1. Fork the repository
2. Create a new branch
3. Fix the issue
4. Submit a pull request to the owner of the original project

If the owner approves your fix, your work should then be merged into the original repository.

## Your idea 

If a project is under public license that allows you to freely use it, you could use the project as a starting point for
your own take on the project. In a scenario where you find a web application that you like, it's under the public license
you can then fork the project and add any additional functionality and it will be yours to freely use or distribute.

## Tasks
### Forking a repository

You will now for a remote repository, set up upstream to get updates from the original and propose a change to the owner

> 1. Ask your trainer to create a repository with some files on it

Reviewer: ```suggestion

```

> 2. Navigate to the repository your trainer created

Reviewer: ```suggestion

```

3. Make sure you're logged in to your github account

> 4. Click on "Fork", this will then create a copy of the repository under your account <br />

Reviewer: ```suggestion
2. Click on "Fork" at the top right of this page for the `notes` project, this will then create a copy of the repository under your account <br />
```
